Automattic/jetpack
# Improve WAF logging

## Proposed changes:

* Adds more information to the WAF logs
* Adds a new toggle that allow users to opt-in to share more detailed debug data such as POST params and request headers for blocked requests.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



Yes. This PR increases the data that we collect for the WAF logs for blocked requests (GET params, user-agent, referer, content-type, request uri) and adds a opt-in toggle to share with us some more additional data (post params and request headers).

## Testing instructions:






* Create a test site with a Scan subscription.
* Install Protect and toggle the share data and share debug data on.
* Make a few requests that are to be blocked.
* Verify that the information was properly logged in the log file.
* Turn off the debug data and verify that the post params and headers aren't being logged.
